### PR TITLE
Fix compilation warnings related to TriCore.

### LINKED ISSUE #2009: tricore warnings
```
arch/TriCore/TriCoreDisassembler.c:188:6: warning: no previous prototype for function 'TriCore_getFeatureBits' [-Wmissing-prototypes]
bool TriCore_getFeatureBits(unsigned int mode, unsigned int feature)
     ^
arch/TriCore/TriCoreDisassembler.c:188:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool TriCore_getFeatureBits(unsigned int mode, unsigned int feature)
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreInstPrinter.c:517:6: warning: no previous prototype for function 'set_mem_access' [-Wmissing-prototypes]
void set_mem_access(MCInst *MI, unsigned int access)
     ^
arch/TriCore/TriCoreInstPrinter.c:517:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
void set_mem_access(MCInst *MI, unsigned int access)
^
static
In file included from arch/TriCore/TriCoreInstPrinter.c:524:
arch/TriCore/TriCoreGenAsmWriter.inc:19:18: warning: no previous prototype for function 'getMnemonic' [-Wmissing-prototypes]
MnemonicBitsInfo getMnemonic(MCInst *MI, SStream *O) {
                 ^
arch/TriCore/TriCoreGenAsmWriter.inc:19:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
MnemonicBitsInfo getMnemonic(MCInst *MI, SStream *O) {
^
static
arch/TriCore/TriCoreGenAsmWriter.inc:3685:6: warning: no previous prototype for function 'printAliasInstr' [-Wmissing-prototypes]
bool printAliasInstr(MCInst *MI, uint64_t Address, SStream *OS) {
     ^
arch/TriCore/TriCoreGenAsmWriter.inc:3685:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool printAliasInstr(MCInst *MI, uint64_t Address, SStream *OS) {
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreModule.c:11:8: warning: no previous prototype for function 'TRICORE_global_init' [-Wmissing-prototypes]
cs_err TRICORE_global_init(cs_struct *ud)
       ^
arch/TriCore/TriCoreModule.c:11:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
cs_err TRICORE_global_init(cs_struct *ud)
^
static
arch/TriCore/TriCoreModule.c:32:8: warning: no previous prototype for function 'TRICORE_option' [-Wmissing-prototypes]
cs_err TRICORE_option(cs_struct *handle, cs_opt_type type, size_t value)
       ^
arch/TriCore/TriCoreModule.c:32:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
cs_err TRICORE_option(cs_struct *handle, cs_opt_type type, size_t value)
^
static
2 warnings generated.
arch/TriCore/TriCoreDisassembler.c:1521:6: warning: no previous prototype for function 'TriCore_getInstruction' [-Wmissing-prototypes]
bool TriCore_getInstruction(csh ud, const uint8_t *code, size_t code_len,
     ^
arch/TriCore/TriCoreDisassembler.c:1521:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool TriCore_getInstruction(csh ud, const uint8_t *code, size_t code_len,
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreDisassembler.c:1570:6: warning: no previous prototype for function 'TriCore_init' [-Wmissing-prototypes]
void TriCore_init(MCRegisterInfo *MRI)
     ^
arch/TriCore/TriCoreDisassembler.c:1570:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
void TriCore_init(MCRegisterInfo *MRI)
^
static
```

### PR DESCRIPTION
**Pull request by b1llow**
Fixes #2009

### CODE REVIEW

**Comment by XVilka** [comment]
@trufae could you please test this one?

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@imbillow if you write `Fixes #xxxx` it will connect to the issue automatically on the GitHub side.

**Comment by trufae** [comment]
Looks good, no warnings for me

**Comment by aquynh** [comment]
dont you want to include `cstool.h` for other archs, too??

File: cstool/cstool.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#ifndef CAPSTONE_CSTOOL_CSTOOL_H_
+#define CAPSTONE_CSTOOL_CSTOOL_H_
+
+void print_insn_detail_x86(csh ud, cs_mode mode, cs_insn *ins);
```
Comment (trufae) [inline cstool/cstool.h:4]: Arent these definitions depending on ifdefs for each selected arch? I think those must be wrapped around ifdefs

Comment (b1llow) [reply cstool/cstool.h:4]: But `cstool` uses `cs_support(CS_ARCH_XXX)` to query capstone for architectural support, not conditional compilation.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@aquynh @kabeor does it look good now?

File: cstool/cstool.h (hunk 1 above)
Comment (kabeor) [reply cstool/cstool.h:4]: Hi, can this thread be marked as solved?

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**